haystack/document_store/elasticsearch.py

```diff
@@ -225,14 +225,46 @@ def get_document_by_id(self, id: str, index=None) -> Optional[Document]:
         else:
             return None
 
-    def get_documents_by_id(self, ids: List[str], index=None) -> List[Document]:
+    def get_documents_by_id(self, ids: List[str], index: Optional[str] = None) -> List[Document]:
         """Fetch documents by specifying a list of text id strings"""
         index = index or self.index
         query = {"query": {"ids": {"values": ids}}}
         result = self.client.search(index=index, body=query)["hits"]["hits"]
         documents = [self._convert_es_hit_to_document(hit, return_embedding=self.return_embedding) for hit in result]
         return documents
 
+    def get_metadata_values_by_key(
+        self,
+        key: str,
+        query: Optional[str] = None,
```

Comment:
What is the use case of query here? Is it to narrow down the search space to fewer documents?

Reply:
Yes, it's for narrowing down to fewer documents. For instance, for a given query, we can get the available meta values.